src/SBarcodeGenerator.cpp

```diff
@@ -112,3 +172,33 @@ void SBarcodeGenerator::setFormat(const QString &formatName)
 {
     setFormat(SCodes::fromString(formatName));
 }
+
+QString SBarcodeGenerator::imagePath() const
+{
+    return m_imagePath;
+}
+
+void SBarcodeGenerator::setImagePath(const QString &imagePath)
+{
+    if (m_imagePath == imagePath) {
+        return;
+    }
+
+    m_imagePath = imagePath;
+    emit imagePathChanged();
+}
+
+int SBarcodeGenerator::centerImageRatio() const
+{
+    return m_centerImageRatio;
+}
+
+void SBarcodeGenerator::setCenterImageRatio(const int &centerImageRatio)
```

Comment:
Do not pass by reference small data types that don't allocate memory when copying. If the size of the type is equal or almost equal the size of the reference.